Refactor register-a-birth to new-style flow

floehopper opened the pull request:
## Description

Trello card: https://trello.com/c/GPLhNzGq

Similar to #2561, this moves the flow towards the style described in the [refactoring documentation][1]. I've concentrated on moving logic out of the flow and into the calculator and presentational concerns into the templates. The idea is to change all the flows to use this approach so we can significantly simplify the DSL.

In this case, I haven't done much work on the calculator itself other than extracting methods from the flow into it. Also I haven't attempted to extract any domain concepts. Some of the logic could almost certainly be simplified. There is still quite a bit of work to do to move all the *tests* for this flow towards the [new-style approach][2].

Although the PR has a lot of commits, the vast majority of them are very small and only affect a few files.

## External changes

None. This is just an internal refactoring.

[1]: https://github.com/alphagov/smart-answers/blob/master/doc/refactoring.md
[2]: https://github.com/alphagov/smart-answers/blob/master/doc/new-style-testing.md

chrisroos commented:
The changes in commit 784b4c10f99699d741e7c666d59ba32b31331ffd (removal of `country_name_query` and changes to register_a_birth_test) appear to be reversed as part of 889742808d0062343f022330072b8c67dd756545 (the next commit). The original changes are then reapplied in 97909720639117cde00e4ce2f3bacaf0feb4b667 and 25a1f39e08ccc067dcf1876e7bc7b324810e2661.

Should the first commit (784b4c10f99699d741e7c666d59ba32b31331ffd) be removed completely?

chrisroos commented:
I've added a single comment about a confusing commit. Aside from that everything looks good to me, @floehopper.

floehopper commented:
I've created a new branch based off this one, removed the commit as suggested by @chrisroos, rebased against master, resolving the conflicts with the recently merged #2610, and opened a new PR: #2619. Closing.
